CoseSignTool/CoseSign1/Extensions/CoseSign1MessageDetachedSignatureExtensions.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace CoseSign1.Extensions;
+
+/// <summary>
+/// Class which extends <see cref="CoseSign1Message"/> for detached signature use cases.
+/// </summary>
+/// <remarks>
+/// Logging is done through Trace.TraceError and Debug.WriteLine.
+/// </remarks>
+public static class CoseSign1MessageDetachedSignatureExtensions
+{
+    private static readonly string AlgorithmGroupName = "algorithm";
+    private static readonly Regex HashMimeTypeExtension = new (@$"(?<extension>\+hash-(?<{AlgorithmGroupName}>\w+))", RegexOptions.Compiled);

```

Comment:
Would you mind adding a comment describing the regex for maintainers who aren't as familiar?